File: solutions/scala/zebra-puzzle/1/src/main/scala/Person.scala
final class Person(
                      val color: Color,
                      val nationality: Nationality,
                      val pet: Pet,
                      val drink: Drink,
                      val hobby: Hobby,
                      val house: House
                  ) {

    def noConflict(other: Person): Boolean = {
        require(other != null, "other")

        (this.color != other.color) &&
            (this.nationality != other.nationality) &&
            (this.pet != other.pet) &&
            (this.drink != other.drink) &&
            (this.hobby != other.hobby) &&
            (this.house != other.house)
    }
}

File: solutions/java/zebra-puzzle/1/src/main/java/Person.java
import java.util.Objects;

public record Person(Color color, Nationality nationality, Pet pet, Drink drink, Hobby hobby, House house) {

    public boolean noConflict(Person other) {
        Objects.requireNonNull(other, "other");
        return this.color != other.color
               && this.nationality != other.nationality
               && this.pet != other.pet
               && this.drink != other.drink
               && this.hobby != other.hobby
               && this.house != other.house;
    }
}